### PR TITLE
[INLONG-4986][Agent] Support MQTT Source

### PR DESCRIPTION
**Pull request by haibo-duan**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-XYZ][Component] Title of the pull request

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- About #4986 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/MqttReader.java [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+                    public void deliveryComplete(IMqttDeliveryToken token) {
+                    }
+                });
+                client.connect(options);
+                client.subscribe(topic, 1);
+            }
+            LOGGER.info("the mqtt subscribe topic is [{}], qos is [{}]", topic, qos);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
```
Comment (GanfengTan) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/MqttReader.java:143]: Suggested to use LOGGER.

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/MqttReader.java [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    }
+
+    private void reconnect() {
+        if (!client.isConnected()) {
+            try {
+                client.connect(options);
+                LOGGER.info("the mqtt client reconnect success");
+            } catch (MqttSecurityException e) {
+                e.printStackTrace();
```
Comment (GanfengTan) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/MqttReader.java:152]: Same as above.

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/MqttReader.java [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+            } catch (MqttException e) {
+                e.printStackTrace();
```
Comment (GanfengTan) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/MqttReader.java:154]: Same as above.

**Comment by dockerzhang** [comment]
@haibo-duan here we need to add MQTT Source to the `JobProfileDto.class`. You can refer to MongoDB in #5674. 
maybe other sources you added in agent also add the job definition.

**Comment by haibo-duan** [comment]
> @haibo-duan here we need to add MQTT Source to the `JobProfileDto.class`. You can refer to MongoDB in #5674. maybe other sources you added in agent also add the job definition.

It has been modified. Please review it for me.

**Review by GanfengTan — APPROVED** [review]

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/MqttReader.java [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+    private int mqttVersion = MqttConnectOptions.MQTT_VERSION_DEFAULT;
+
+    private LinkedBlockingQueue<DefaultMessage> mqttMessagesQueue;
+
+    public MqttReader(String topic) {
+        this.topic = topic;
+    }
+
+    @Override
+    public void init(JobProfile jobConf) {
```
Comment (EMsnap) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/MqttReader.java:83]: method is way too long

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/pojo/JobProfileDto.java [hunk 5]
```diff
@@ -54,6 +54,10 @@ public class JobProfileDto {
      * mongo source
      */
     public static final String MONGO_SOURCE = "org.apache.inlong.agent.plugin.sources.MongoDBSource";
+    /**
+     * mqtt source
+     */
+    public static final String MQTT_SOURCE = "org.apache.agent.plugin.sources.MqttSource";
```
Comment (pocozh) [inline inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/pojo/JobProfileDto.java:60]: wrong package name

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/MqttReader.java [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+        options.setAutomaticReconnect(automaticReconnect);
+        options.setMqttVersion(mqttVersion);
+
+        try {
+            synchronized (MqttReader.class) {
+                client = new MqttClient(serverURI, clientId, new MemoryPersistence());
+                client.setCallback(new MqttCallback() {
+                    @Override
+                    public void connectionLost(Throwable cause) {
+                        LOGGER.info("the mqtt connection is lost, try to reconnect");
```
Comment (pocozh) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/MqttReader.java:119]: It's better to add  some necessary info such as jobId, connection etc in this log.

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/MqttReader.java [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+    @Override
+    public void setReadTimeout(long mill) {
+    }
+
+    @Override
+    public void setWaitMillisecond(long millis) {
+    }
+
+    @Override
+    public String getSnapshot() {
```
Comment (pocozh) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/MqttReader.java:205]: Could you please complete the strategy of snapshot?

Comment (haibo-duan) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/MqttReader.java:207]: @pocozh   when the mqtt topic is subscribed, the message is pushed to the client through the broker. There is no offset attribute, and the snapshot may not be necessary.

**Review by EMsnap — APPROVED** [review]

**Review by pocozh — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**